marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallbackGeneratorTest.java

```diff
@@ -0,0 +1,415 @@
+package com.marklogic.client.test.dataservices;
+
+import static org.hamcrest.core.StringStartsWith.startsWith;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.util.Date;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.transform.Source;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.MarkLogicInternalException;
+import com.marklogic.client.dataservices.CallBatcher.CallArgsGenerator;
+import com.marklogic.client.dataservices.CallManager;
+import com.marklogic.client.dataservices.CallManager.CallArgs;
+import com.marklogic.client.dataservices.impl.CallBatcherImpl;
+import com.marklogic.client.dataservices.impl.CallBatcherImpl.BuilderImpl;
+import com.marklogic.client.dataservices.impl.CallManagerImpl.CallerImpl;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+import com.marklogic.client.test.Common;
+
+public class CallbackGeneratorTest implements CallArgsGenerator<CallManager.CallEvent>{
+
+    private static DatabaseClient db = Common.connect();
+    private static CallerImpl callerImpl;
+    private static CallbackGeneratorTest callbackGeneratorTest;
+    
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void setup() {
+        callerImpl = new CallerImpl<CallManager.CallEvent>(null) {
+
+            @Override
+            public CallManager.CallEvent callForEvent(DatabaseClient client, CallArgs args) throws Exception {
+                return null;
+            }
+        };
+        callbackGeneratorTest = new CallbackGeneratorTest();
+    }
+
+    @Test(expected = Test.None.class)
+    public void forArgsGeneratorTest() {
+        BuilderImpl builderImpl = new BuilderImpl(db,callerImpl);
+        assertNotNull(builderImpl.forArgsGenerator(callbackGeneratorTest));
+    }
+    
+    @Test
+    public void voidInputTypeTest() {
```

Comment:
This is a good test to capture -- it would just be necessary to rework it with the public interfaces instead of the implementation.